File: sort/reverseList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
		
		Input: 1->2->3
		Output: 3->2->1
        """
        prev, curr = None, head
        while curr:
            temp = curr
            curr = curr.next
            temp.next = prev
            prev = temp
        return prev

File: sort/reverseList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode prev = new ListNode(0);
        prev = prev.next;
        ListNode curr = head;
        while(curr != null) {
            ListNode temp = curr;
            curr = curr.next;
            temp.next = prev;
            prev = temp;
        }       
        return prev;
    }
}